.github/workflows/build-and-deploy.yml

```diff
@@ -50,14 +52,14 @@ jobs:
       AWS_DEFAULT_REGION: us-east-1
     steps:
     - name: build/checkout-repo
-      uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
```

Comment:
This step is redundant .

Reply:
let's remove it then

Reply:
@toninis On a second thought, I'll merge first, I need to send more PRs anyway otherwise as soon as I make a change here I'll invalidate reviews once more and start from scratch.